src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Resources/BaseConditionalHandlerTests.cs

```diff
@@ -0,0 +1,163 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Hl7.Fhir.Model;
+using Microsoft.Health.Fhir.Core.Exceptions;
+using Microsoft.Health.Fhir.Core.Extensions;
+using Microsoft.Health.Fhir.Core.Features.Conformance;
+using Microsoft.Health.Fhir.Core.Features.Persistence;
+using Microsoft.Health.Fhir.Core.Features.Resources;
+using Microsoft.Health.Fhir.Core.Features.Search;
+using Microsoft.Health.Fhir.Core.Models;
+using Microsoft.Health.Fhir.Shared.Core.UnitTests.Features.Resources;
+using Microsoft.Health.Fhir.Tests.Common;
+using NSubstitute;
+using Xunit;
+using static Hl7.Fhir.Model.Bundle;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.Health.Fhir.Core.UnitTests.Features.Resources
+{
+    public class BaseConditionalHandlerTests
+    {
+        private readonly TestBaseConditionalHandler _testBaseConditionalHandler;
+        private readonly ISearchService _searchService = Substitute.For<ISearchService>();
+
+        public BaseConditionalHandlerTests()
+        {
+            IFhirDataStore fhirDataStore = Substitute.For<IFhirDataStore>();
+            Lazy<IConformanceProvider> conformanceProvider = Substitute.For<Lazy<IConformanceProvider>>();
+            IResourceWrapperFactory resourceWrapperFactory = Substitute.For<IResourceWrapperFactory>();
+            ResourceIdProvider resourceIdProvider = Substitute.For<ResourceIdProvider>();
+            _testBaseConditionalHandler = new TestBaseConditionalHandler(fhirDataStore, _searchService, conformanceProvider, resourceWrapperFactory, resourceIdProvider);
+        }
+
+        [Fact]
+        public async Task GivenATransactionBundleWithIdentifierReferences_WhenResolved_ThenReferencesValuesAreNotUpdated()
+        {
```

Comment:
I donot think, this is a right class for these bundle specific tests.

Reply:
These tests are to validate the behavior of resolveRefereneAsSync method and now this method is moved to BaseConditionalHandler, So I think its good to have unit tests in the corresponding test class ie. BaseConditionalHandlerTests. Thoughts?

Reply:
I guess it makes sense. Since all these tests are using bundle as a resource, I was thinking if it will make sense to put these separately. But there is no harm to leave them here too.